.github/workflows/process_data.py

```diff
@@ -51,105 +51,174 @@ def get_media(item_id):
-        media.get("o:title")
-        if index is not None
-        else data.get("dcterms:title", [{}])[0].get("@value")
+
+
+def extract_prop_uri(props, prop_id):
+    return next(
+        (
+            prop.get("@id", "")
+            if "o:label" not in prop
+            else f"[{prop.get('o:label', '')}]({prop.get('@id', '')})"
+            for prop in props
+            if prop.get("property_id") == prop_id
+        ),
+        "",
     )
 
+
+def extract_combined_list(props):
+    values = [prop.get("@value", "") for prop in props if "@value" in prop]
+    uris = [
+        prop.get("@id", "")
+        if "o:label" not in prop
+        else f"[{prop.get('o:label', '')}]({prop.get('@id', '')})"
+        for prop in props
+        if "@id" in prop
+    ]
+    combined = values + uris
+    return ";".join(combined)
+
+
+def extract_item_data(item):
+    image_url = item.get("thumbnail_display_urls", {}).get("large", "")
+    local_image_path = ""
+    if image_url:
+        local_image_path = f"objects/{item['o:id']}.jpg"
+        download_file(image_url, local_image_path)
+
     return {
-        "objectid": data.get("dcterms:identifier", [{}])[0].get("@value").lower()
-        + media_id_suffix,
-        "parentid": None
-        if index is None
-        else data.get("dcterms:identifier", [{}])[0].get("@value").lower(),
-        "title": media_title,
-        "creator": data.get("dcterms:creator", [{}])[0].get("@value"),
-        "date": data.get("dcterms:date", [{}])[0].get("@value"),
-        "era": data.get("dcterms:temporal", [{}])[0].get("@value"),
-        "description": data.get("dcterms:description", [{}])[0].get("@value"),
-        "subject": "; ".join(
-            [item.get("@value", "") for item in data.get("dcterms:subject", [])]
-        ),
-        "publisher": data.get("dcterms:publisher", [{}])[0].get("@value"),
-        "source": data.get("dcterms:source", [{}])[0].get("@value"),
-        "relation": data.get("dcterms:relation", [{}])[0].get("@value"),
-        "hasVersion": data.get("dcterms:hasVersion", [{}])[0].get("@value"),
-        "type": data.get("dcterms:type", [{}])[0].get("@id"),
-        "format": data.get("dcterms:format", [{}])[0].get("@value"),
-        "extent": data.get("dcterms:extent", [{}])[0].get("@value"),
-        "language": data.get("dcterms:language", [{}])[0].get("o:label"),
-        "rights": "; ".join(
-            [item.get("@value", "") for item in data.get("dcterms:rights", [])]
-        ),
-        "license": data.get("dcterms:license", [{}])[0].get("@value"),
-        "isPartOf": data.get("dcterms:isPartOf", [{}])[0].get("@value"),
-        "isReferencedBy": data.get("dcterms:isReferencedBy", [{}])[0].get("@value"),
-        "display_template": type,
-        "object_location": media_original_url,
-        "image_small": media_url,
-        "image_thumb": media_url,
-        "image_alt_text": None,
-        "object_transcript": None,
+        "objectid": item["o:id"],
```

Comment:
you are using the omeka internal id here. but it needs "our" id (dcterms:identifier)
```suggestion
        "objectid": extract_prop_value(item.get("dcterms:identifier", []), 10),
```
but of course this has consequences for line 201